Add oauth-proxy imagestream

IsaiahStapleton opened the pull request:
This image is used by RHOAI to provide authentication. RHOAI is currently pulling this image externally but we would like to store it and pull from the internal image repo.

computate reviewed (commented):
@IsaiahStapleton Why are we introducing the exact image that already exists? 

```bash
$ oc -n redhat-ods-applications get imagestream/oauth-proxy -o yaml | grep from: -A 2
    from:
      kind: DockerImage
      name: registry.redhat.io/openshift4/ose-oauth-proxy@sha256:4bef31eb993feb6f1096b51b4876c65a6fb1f4401fee97fa4f4542b6b7c9bc46
```

It doesn't seem like a good idea to hard-code the image with this sha256. I imagine OpenShift automatically updates this over time. Maybe I don't understand the goals of this. Maybe you could link it to an issue.

larsks reviewed (changes requested):
This needs to be included in a kustomization file.

IsaiahStapleton committed:
Add oauth-proxy imagestream

This image is used by RHOAI to provide authentication. RHOAI
is currently pulling this image externally but we would like
to store it and pull from the internal image repo.

Signed-off-by: Isaiah Stapleton <istaplet@redhat.com>
